cleverhans/attacks_tf.py

```diff
@@ -1918,3 +1918,223 @@ def margin_logit_loss(model_logits, label, num_classes=10):
         logits_with_target_label_neg_inf, axis=-1)
     loss = highest_nonlabel_logits - label_logits
     return loss
+
+
+def _transformer(U, theta, out_size, name='SpatialTransformer', **kwargs):
```

Comment:
I have trouble following what's going on in the `_transformer` method.

Is it doing something similar to the transform method from the original Engstrom et al. paper?
([our slightly modified version](https://github.com/google/unrestricted-adversarial-examples/blob/1e38f25ec4aee5d5105580f3501335e792a837dd/unrestricted-advex/unrestricted_advex/attacks.py#L240) copied below) 
```
def apply_transformation(x, transform, image_height, image_width):
  # Map a transformation onto the input
  trans_x, trans_y, rot = tf.unstack(transform, axis=1)
  rot *= np.pi / 180  # convert degrees to radians

  # Pad the image to prevent two-step rotation / translation
  # resulting in a cropped image
  x = tf.pad(x, [[0, 0], [50, 50], [50, 50], [0, 0]], 'CONSTANT')

  # rotate and translate image
  ones = tf.ones(shape=tf.shape(trans_x))
  zeros = tf.zeros(shape=tf.shape(trans_x))
  trans = tf.stack([ones, zeros, -trans_x,
                    zeros, ones, -trans_y,
                    zeros, zeros], axis=1)
  x = tf.contrib.image.rotate(x, rot, interpolation='BILINEAR')
  x = tf.contrib.image.transform(x, trans, interpolation='BILINEAR')
  return tf.image.resize_image_with_crop_or_pad(
    x, image_height, image_width)
```

Reply:
If so, maybe use the `tf.contrib.image` library?

If not, perhaps you could add tests for the `_transformer` method to show the usage and correctness?